cluster-autoscaler/proposals/clusterstate.md

```diff
@@ -0,0 +1,58 @@
+### Cluster State Registry
+### Handling unready nodes 
+
+### Introduction
+
+Currently ClusterAutoscaler stops working when the number of nodes observed on the cloud provider side doesn’t match to the number of ready nodes on the Kubernetes side. This behavior was introduced in the first version of CA in order to prevent CA from breaking more the already broken cluster. However the client reported feedback indicated that this behavior is oftentimes suboptimal and leads to confusion. This document sketches how the problem of unready nodes will be solved in the next release of K8S.
+
+### Use cases
+
+The number of ready nodes can be different than on the mig side in the following situations:
+
+* [UC1] A new node is being added to the cluster. The node group has been increased but the node has not been created/started/registered in K8S yet. On GCP this usually takes couple minutes. 
+Indicating factors:
+ -- There was a scale up in the last couple minutes.
+ -- The number of missing node is at most the size of executed scale-up.
+Suggested action: Continue operations, however include all yet-to-arrive nodes in all scale-up considerations.
+
+* [UC2] A new node is being added to the cluster. The node has registered on the cluster but has not yet switched its state to ready. This should be fixed in couple seconds. Indicating factors:
+ -- The unready node is new. CreateTime in the last couple minutes.
+Suggested action: Continue operations, however include all yet-to-arrive nodes in all scale-up considerations.
+
+* [UC3] A new node was added to the cluster but failed to start within the reasonable time. There is little chance that it will start anytime soon. Indicating factors:
+ -- Node is unready
+ -- CreateTime == unready NodeCondition.LastTransitionTime
+Suggested action: Continue operations, however do not expand this node pool. The probable scenario is that the node will be picked by scale down soon.
```

Comment:
Shouldn't w add a condition here that the node is unready for a long enough time?